src/lodepng/lodepng.cpp

```diff
@@ -1250,33 +1250,28 @@ static void getPixelColorRGBA8(unsigned char* r, unsigned char* g,
   if(mode->colortype == LCT_GREY) {
     if(mode->bitdepth == 8) {
       *r = *g = *b = in[i];
-      if(mode->key_defined && *r == mode->key_r) *a = 0;
-      else *a = 255;
+      *a = 255 * !(mode->key_defined && *r == mode->key_r);
```

Comment:
Some of these changes are not a good fit for ECT as I'm trying to keep the differences with mainline lodepng small. Since lodepng is not responsible for much of ECT’s runtime for most use cases, small improvements to lodepng code size or run time will not make much of a difference while making it harder to maintain. 
In particular, the changes where you introduce multiplication may be more difficult to read/understand and adding const statements should not really affect code generation, so I will keep the upstream version.  If you feel like a change makes a meaningful difference, you can also try to submit a pull request to mainline lodepng.  I will review the changes to the ECT-specific code (e.g. the changes to filtering) later, although I’m currently swamped with school work so it might be a while.